### ISSUE TITLE
Using full-install command fails due to no valid magic files.

### ISSUE THREAD

**Issue by pl232602** [Mord3rca/gamma-launcher#113]
System: 
OS: Arch Linux x86_64 
Host: B550M AORUS ELITE -CF 
Kernel: 6.8.7-arch1-2
Shell: bash 5.2.26 
Resolution: 1920x1080 
DE: GNOME 46.1
CPU: AMD Ryzen 5 5600X (12) @ 3.700GHz 
GPU: NVIDIA GeForce GTX 1060 3GB

Issue: When running the command ./gamma-launcher full-install, it will download for a few seconds before giving the following error:
```
/etc/magic, 0: Warning: using regular magic file `/usr/share/misc/magic'
/etc/magic, 0: Warning: using regular magic file `/usr/share/misc/magic'
[+] Updating G.A.M.M.A. definition
    will be updated from 840 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [00:55, 5.70MiB/s] 
/etc/magic, 0: Warning: using regular magic file `/usr/share/misc/magic'
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 60, in extract_archive
  File "magic/__init__.py", line 178, in from_file
  File "magic/__init__.py", line 165, in _get_magic_type
  File "magic/__init__.py", line 73, in __init__
  File "magic/__init__.py", line 332, in magic_load
  File "magic/__init__.py", line 225, in errorcheck_negative_one
magic.MagicException: b'could not find any valid magic files!'
[3771] Failed to execute script 'gamma-launcher' due to unhandled exception!
```

**Comment by Mord3rca** [comment]
Hey,
is `file` package installed on your system ?

**Comment by pl232602** [comment]
Yes, its installed.

**Comment by Mord3rca** [comment]
Can you give me the output of `file --version` ?

**Comment by pl232602** [comment]
```
file-5.45
magic file from /usr/share/file/misc/magic
seccomp support included
```

**Comment by Mord3rca** [comment]
Great, then can you try:
`MAGIC=/usr/share/file/misc/magic ./gamma-launcher full-install [...]` ?

**Comment by pl232602** [comment]
I ran that and this happened:
```
[+] Updating G.A.M.M.A. definition
    will be updated from 840 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [01:08, 4.63MiB/s] 
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 60, in extract_archive
  File "magic/__init__.py", line 178, in from_file
  File "magic/__init__.py", line 165, in _get_magic_type
  File "magic/__init__.py", line 73, in __init__
  File "magic/__init__.py", line 332, in magic_load
  File "magic/__init__.py", line 225, in errorcheck_negative_one
magic.MagicException: b"File 5.41 supports only version 16 magic files. `/usr/share/file/misc/magic.mgc' is version 18"
[5870] Failed to execute script 'gamma-launcher' due to unhandled exception!
```

**Comment by Mord3rca** [comment]
Progress ! So, this is because the libfile.so in the self extract executable is too old for your system.

So 2 solutions:
* Use venv / pip install
* Try again with `LD_PRELOAD=/path/to/libfile.so/on/your/system`

**Comment by JamesMowery** [comment]
Can confirm this issue on both NixOS and Debian 12 as well. None of the solutions above worked.

```
(env) james@phoenix:/mnt/games/gamma-launcher$ ./gamma-launcher full-install --anomaly /mnt/games/Anomaly/ --gamma /mnt/games/GAMMA
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
[+] Updating G.A.M.M.A. definition
    will be updated from 843 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [00:10, 29.7MiB/s]
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 61, in extract_archive
TypeError: 'NoneType' object is not callable
[4395] Failed to execute script 'gamma-launcher' due to unhandled exception!
(env) james@phoenix:/mnt/games/gamma-launcher$ pip install python-magic
Collecting python-magic
  Downloading python_magic-0.4.27-py2.py3-none-any.whl.metadata (5.8 kB)
Downloading python_magic-0.4.27-py2.py3-none-any.whl (13 kB)
Installing collected packages: python-magic
Successfully installed python-magic-0.4.27
(env) james@phoenix:/mnt/games/gamma-launcher$ ./gamma-launcher full-install --anomaly /mnt/games/Anomaly/ --gamma /mnt/games/GAMMA
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
[+] Updating G.A.M.M.A. definition
    will be updated from 843 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [00:09, 34.6MiB/s]
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 61, in extract_archive
TypeError: 'NoneType' object is not callable
[4418] Failed to execute script 'gamma-launcher' due to unhandled exception!
(env) james@phoenix:/mnt/games/gamma-launcher$ MAGIC=/usr/share/file/magic/ ./gamma-launcher full-install --anomaly /mnt/games/Anomaly/ --gamma /mnt/games/GAMMA
[+] Updating G.A.M.M.A. definition
    will be updated from 843 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [00:09, 34.0MiB/s]
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 61, in extract_archive
TypeError: 'NoneType' object is not callable
[4469] Failed to execute script 'gamma-launcher' due to unhandled exception!
(env) james@phoenix:/mnt/games/gamma-launcher$ LD_PRELOAD=/usr/lib/libmagic.so ./gamma-launcher full-install --anomaly /mnt/games/Anomaly/ --gamma /mnt/games/GAMMA
ERROR: ld.so: object '/usr/lib/libmagic.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/libmagic.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
[+] Updating G.A.M.M.A. definition
    will be updated from 843 to 846
    downloading archive...
  - Downloading Stalker_GAMMA-main.zip: 302MiB [00:09, 32.9MiB/s]
/etc/magic, 4: Warning: using regular magic file `/usr/share/misc/magic'
Traceback (most recent call last):
  File "gamma-launcher.py", line 8, in <module>
  File "launcher/cli.py", line 68, in main
  File "launcher/commands/install.py", line 341, in run
  File "launcher/commands/install.py", line 258, in _update_gamma_definition
  File "launcher/archive.py", line 76, in extract_git_archive
  File "launcher/archive.py", line 61, in extract_archive
TypeError: 'NoneType' object is not callable
[4480] Failed to execute script 'gamma-launcher' due to unhandled exception!
```

**Comment by JamesMowery** [comment]
Also another clear issue with this is that the full-install command continuously downloads 300MiB every single attempt prior to breaking. I think this places unnecessary load on the servers. This check should be done before downloading any files to be more bandwidth friendly.

**Comment by Mord3rca** [comment]
@JamesMowery This won't happen if you use a cache with `--cache-directory` option

Also, binary are made on Ubuntu and are probably 100% compatible only with this distro.

**Comment by pl232602** [comment]
LD_PRELOAD had the same result but the venv method worked without issue.

**Comment by JamesMowery** [comment]
@Mord3rca that makes sense! Would you be so kind to update the readme to note that the release version should prefer to use Ubuntu (if it was already mentioned, I totally missed it).

 Currently installing the game via clone.  In the middle of upgrading. Hoping it works!

**Comment by JamesMowery** [comment]
So the upgrade completely broke my install. I decided to nuke my GAMMA folder and start fresh. Seemed to be going well, but at some point I got this error:

```
[+] Installing mod: Black Jack
  - Downloading 21_game_v1.1.4.1.7z: 6.13MiB [00:00, 15.3MiB/s]
    Installing CORE -> /mnt/games/GAMMA/mods/28- Black Jack - Feel_Fried
[+] Installing mod: High Res PDA Maps
  - Downloading High_Resolution_PDA_Maps.1.zip: 116MiB [00:01, 93.0MiB/s]
    Installing High Res PDA Maps -> /mnt/games/GAMMA/mods/26- High Res PDA Maps - Bazingarrey
    WARNING: High Res PDA Maps does not exist
[+] Installing mod: High Res Loading Screens
  - Downloading HighResLoadingScreenV1.1.7z: 105MiB [00:01, 91.5MiB/s]
[+] Installing mod: Skies Redux
Traceback (most recent call last):
  File "/mnt/games/gamma-launcher/gamma-launcher/env/bin/gamma-launcher", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/cli.py", line 68, in main
    args.cobject().run(args)
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/commands/install.py", line 345, in run
    self._install_mods()
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/commands/install.py", line 296, in _install_mods
    mod.install(self._dl_dir, self._mod_dir)
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/mods/base.py", line 192, in install
    archive = self.download(download_dir)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/mods/base.py", line 176, in download
    return download_archive(self._url, mod_dir)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/tenacity/__init__.py", line 289, in wrapped_f
    return self(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/tenacity/__init__.py", line 379, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/tenacity/__init__.py", line 314, in iter
    return fut.result()
           ^^^^^^^^^^^^
  File "/usr/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/tenacity/__init__.py", line 382, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/downloader/__init__.py", line 38, in download_archive
    file = Path(download_dir, e.filename)
                              ^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/downloader/moddb.py", line 51, in filename
    self._get_filename()
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/launcher/downloader/moddb.py", line 60, in _get_filename
    location = g_session.get(f"https://www.moddb.com{s[0]}", allow_redirects=False).headers["location"]
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/mnt/games/gamma-launcher/gamma-launcher/env/lib/python3.11/site-packages/requests/structures.py", line 52, in __getitem__
    return self._store[key.lower()][1]
           ~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'location'
```

**Comment by JamesMowery** [comment]
Nice dice. I setup everything from scratch. The game just does not want to detect the mods. And this was never, ever a problem before. I followed the instructions. Not really sure what's going on.

![image](https://github.com/Mord3rca/gamma-launcher/assets/868178/65672496-20bd-4a96-b924-b32a210fc1fa)

**Comment by Mord3rca** [comment]
@JamesMowery Do you have another profile for ModOrganizer ? Default is not valid, GAMMA need to be selected.

**Comment by JamesMowery** [comment]
@Mord3rca I attempted to create a new profile because the first one didn't take. Is there a way to delete all the profiles and start over to where it asks you to select the Anomaly folder?

**Comment by Mord3rca** [comment]
Delete *ModOrganizer.ini* and remove *profiles/default*

Yes, the splash is the default one, this is expected

**Comment by JamesMowery** [comment]
@Mord3rca I do appear to have a G.A.M.M.A profile, and no default one. But the GAMMA profile did not show up in MO. Should I delete the GAMMA folder and the ModOrganizer.ini?

![image](https://github.com/Mord3rca/gamma-launcher/assets/868178/182ef6d5-2a46-4913-bce5-443710396dd8)

**Comment by JamesMowery** [comment]
@Mord3rca I think I know what's going on. There seems to be a disconnect between the Lutris Wineprefix location and where GAMMA is installing the files. I was able to locate the G.A.M.M.A and Default folders in a completely different places on my system.

`/mnt/games/Prefixes/GAMMA/drive_c/users/steamuser/AppData/Local/ModOrganizer/STALKER Anomaly` (which only has a Default folder, I believe this is my prefix)
vs
`/mnt/games/GAMMA/profiles` (which has the G.A.M.M.A folder)

Not really sure how to go about solving this. Maybe the Lutris setup is failing?

**Comment by JamesMowery** [comment]
Okay, after randomly deleting some stuff, I managed to get further. It was a really odd situation of having to delete profiles and stuff. Regardless, the mods are now showing up, which is great!

The problem is that I'm getting this error now:

![image](https://github.com/Mord3rca/gamma-launcher/assets/868178/49f33f56-d050-4d4f-a910-da54269483ab)

**Comment by pl232602** [comment]
Set the anomaly directory as your working directory, that solved my post-install issues.

**Comment by JamesMowery** [comment]
@pl232602 Yeah that's how I actually got to finally get into the game menu. Unfortunately, after I attempt to load into the game world, I'm still getting various fatal errors. Seems to happen at random. Once or twice I was able to actually make it in game, but I could only see a completely black screen but could see the UI elements.

![image](https://github.com/Mord3rca/gamma-launcher/assets/868178/d67c8054-c442-41f7-883a-98b24d56bfc2)

**Comment by JamesMowery** [comment]
@Mord3rca both good and sad to know it's not just me then. I'll keep watching that issue!

One last question. If we want to update the game, we should just run the ./gamma-launcher full-install command? And it'll automatically update? That's what I was previously doing (and I unfortunately always had failures after that; still sad I nuked my only working attempt).

**Comment by Mord3rca** [comment]
TL;DR: Error is caused by libmagic missmatch between ubuntu (in executable) / arch.
Solution: Use venv to use system library
